### PR TITLE
Add support to using more than one MPI rank per ensemble member

### PR DESCRIPTION
**Pull request by Rombur**
This PR enables us to use multiple processors per ensemble member. This only works if the all the ensemble members use the same number of processors. This ensures that the mesh is partitioned the same way for all the ensemble members which simplifies some operations. We could remove this condition in the future.

The changes in the code are pretty small if you hide the whitespace changes. Most of the bugs were related to some data not being distributed to all the processors.

### CODE REVIEW

**Comment by Rombur** [comment]
This requires #268. Otherwise we get too many warnings and the compilation stops

**Comment by stvdwtt** [comment]
@Rombur, got it, I just merged in #268.

**Comment by Rombur** [comment]
Retest this please

**Comment by Rombur** [comment]
Retest this please

File: ci/jenkins_config [hunk 1]
```diff
@@ -38,7 +38,7 @@ pipeline {
                   -D DEAL_II_DIR=${DEAL_II_DIR} \
                   ..
                 '''
-                sh 'make -j3'
+                sh 'make -j2'
```
Comment (Rombur) [inline ci/jenkins_config:41]: We need a little bit less than 3GB per process but the CPU machine only have 8GB of RAM. Reducing the number of processors fixed the compilation error.

File: ci/jenkins_config [hunk 2]
```diff
@@ -22,7 +22,7 @@ pipeline {
             docker {
               image "rombur/adamantine-stack:no_gpu-latest"
                 alwaysPull true
-                label 'docker || nvidia-docker || rocm-docker'
+                label 'nvidia-docker || rocm-docker'
```
Comment (Rombur) [inline ci/jenkins_config:25]: The CPU nodes don't have enough memory and cores. Instead use a GPU node

**Review by stvdwtt — APPROVED** [review]

**Pull request merged**